waggle-dance-core/src/main/java/com/hotels/bdp/waggledance/mapping/service/impl/PrefixBasedDatabaseMappingService.java

```diff
@@ -80,32 +80,26 @@ public PrefixBasedDatabaseMappingService(
     init(initialMetastores);
   }
 
-  private void init(List<AbstractMetaStore> federatedMetaStores) {
+  private void init(List<AbstractMetaStore> metaStores) {
     mappingsByPrefix = Collections.synchronizedMap(new LinkedHashMap<>());
     mappedDbByPrefix = new ConcurrentHashMap<>();
-    for (AbstractMetaStore federatedMetaStore : federatedMetaStores) {
-      add(federatedMetaStore);
+    for (AbstractMetaStore abstractMetaStore : metaStores) {
+      add(abstractMetaStore);
     }
   }
 
-  private void add(AbstractMetaStore federatedMetaStore) {
-    MetaStoreMapping metaStoreMapping = metaStoreMappingFactory.newInstance(federatedMetaStore);
+  private void add(AbstractMetaStore metaStore) {
+    MetaStoreMapping metaStoreMapping = metaStoreMappingFactory.newInstance(metaStore);
 
-    if (federatedMetaStore.getFederationType() == PRIMARY) {
+    if (metaStore.getFederationType() == PRIMARY) {
```

Comment:
I am not exactly sure how `primaryDatabaseMapping` is used in rest of the WD code but perhaps this can be re-written like this:

```
 DatabaseMapping databaseMapping = createDatabaseMapping(metaStoreMapping);

    if (metaStore.getFederationType() == PRIMARY) {
      primaryDatabaseMapping = databaseMapping;
    }
    mappingsByPrefix.put(metaStoreMapping.getDatabasePrefix(), databaseMapping);
```

Reply:
You're right, that would be much simpler; thanks :)
It's done now